src/Core/Settings.h

```diff
@@ -726,7 +726,7 @@ static constexpr UInt64 operator""_Gb(unsigned long long value)
     M(Bool, output_format_pretty_row_numbers, false, "Add row numbers before each row for pretty output format", 0) \
     M(Bool, insert_distributed_one_random_shard, false, "If setting is enabled, inserting into distributed table will choose a random shard to write when there is no sharding key", 0) \
     \
-    M(Bool, cross_to_inner_join_rewrite, true, "Use inner join instead of comma/cross join if possible", 0) \
+    M(UInt64, cross_to_inner_join_rewrite, 1, "Use inner join instead of comma/cross join if possible. Possible values: 0 - no rewrite, 1 - apply if possible, 2 - force rewrite all cross joins", 0) \
```

Comment:
Maybe better to introduce new enum for it? (see SettingsEnums.h)

Reply:
But we still should support values `cross_to_inner_join_rewrite = 1` and `cross_to_inner_join_rewrite = 0`, how to make it backward compatible?

Reply:
Need to check if it would work:
```
IMPLEMENT_SETTING_ENUM(RewriteMode, ErrorCodes::BAD_ARGUMENTS,
    {{"1", RewriteMode::Enable},
     {"0", RewriteMode::Disable},
     {"true", RewriteMode::Enable},
     {"True", RewriteMode::Enable},
     {"TRUE", RewriteMode::Enable},
     {"false", RewriteMode::Disable},
     {"False", RewriteMode::Disable},
     {"FALSE", RewriteMode::Disable},
     {"enable", RewriteMode::Enable},
     {"disable", RewriteMode::Disable},
     {"force", RewriteMode::Force}})
```

But looks a bit overkill

Reply:
Yes, it's an overkill, agree.